programs/bpf_loader/src/syscalls/mod.rs

```diff
@@ -1710,6 +1716,92 @@ declare_syscall!(
     }
 );
 
+declare_syscall!(
+    /// Big integer modular exponentiation
+    SyscallBigModExp,
+    fn inner_call(
+        invoke_context: &mut InvokeContext,
+        params: u64,
+        return_value: u64,
+        _arg3: u64,
+        _arg4: u64,
+        _arg5: u64,
+        memory_mapping: &mut MemoryMapping,
+    ) -> Result<u64, EbpfError> {
+        let params = &translate_slice::<BigModExpParams>(
+            memory_mapping,
+            params,
+            1,
+            invoke_context.get_check_aligned(),
+            invoke_context.get_check_size(),
+        )?
+        .get(0)
+        .ok_or(SyscallError::InvalidLength)?;
+
+        let input_len: u64 = std::cmp::max(params.base_len, params.exponent_len);
+        let input_len: u64 = std::cmp::max(input_len, params.modulus_len);
+
+        let budget = invoke_context.get_compute_budget();
+        consume_compute_meter(
+            invoke_context,
+            budget.syscall_base_cost.saturating_add(
+                budget
+                    .big_modular_exponentiation_cost
+                    .saturating_mul(input_len),
+            ),
+        )?;
+
+        let base = translate_slice::<u8>(
+            memory_mapping,
+            params.base as *const _ as *const u8 as u64,
+            params.base_len,
+            invoke_context.get_check_aligned(),
+            invoke_context.get_check_size(),
+        )?;
+
+        let exponent = translate_slice::<u8>(
+            memory_mapping,
+            params.exponent as *const _ as *const u8 as u64,
+            params.exponent_len,
+            invoke_context.get_check_aligned(),
+            invoke_context.get_check_size(),
+        )?;
+
+        let modulus = translate_slice::<u8>(
+            memory_mapping,
+            params.modulus as *const _ as *const u8 as u64,
+            params.modulus_len,
+            invoke_context.get_check_aligned(),
+            invoke_context.get_check_size(),
+        )?;
+
+        let base = BigUint::from_bytes_be(base);
+        let exponent = BigUint::from_bytes_be(exponent);
+        let modulus = BigUint::from_bytes_be(modulus);
+
+        let value = if modulus.is_zero() || modulus.is_one() {
```

Comment:
Instead of defining the exponentiation operation here, can we call the `big_mod_exp` in the sdk?

Reply:
fixed